x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/helpers.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AssetCriticalityRecord } from '../../../../common/api/entity_analytics';
+import { CriticalityModifiers } from './constants';
+
+/**
+ * Retrieves the criticality modifier for a given criticality level.
+ *
+ * @param criticalityLevel The criticality level for which to get the modifier.
+ * @returns The associated criticality modifier for the given criticality level.
+ */
+export const getCriticalityModifier = (
+  criticalityLevel?: AssetCriticalityRecord['criticality_level']
+): number | undefined => {
+  if (criticalityLevel == null) {
+    return;
+  }
+
+  return CriticalityModifiers[criticalityLevel];
+};
+
+/**
+ * Applies asset criticality to a normalized risk score using bayesian inference.
+ * @param modifier - The criticality modifier to apply to the score.
+ * @param score - The normalized risk score to which the criticality modifier is applied
+ *
+ * @returns The risk score with the criticality modifier applied.
+ */
+export const applyCriticalityToScore = ({
+  modifier,
+  score,
+}: {
+  modifier: number | undefined;
+  score: number;
+}): number => {
+  if (modifier == null) {
+    return score;
+  }
+
+  return bayesianUpdate({ max: 100, modifiers: [modifier], score });
+};
+
+/**
+ * Updates a score with the given modifiers using bayesian inference.
+ * @param modifiers - The modifiers to be applied to the score.
+ * @param score - The score to modifiers are applied
+ * @param max - The maximum value of the score.
+ *
+ * @returns The updated score with modifiers applied
+ */
+export const bayesianUpdate = ({
```

Comment:
`bayesianApplyModifier` maybe?

Reply:
FWIW I got the term "bayesian update" from [this page](https://en.wikipedia.org/wiki/Bayesian_inference); I'm not really familiar with any of this math terminology anymore 😆 .

With the JSDoc I'm not super concerned about the name; happy to change it to whatever is most straightforward. 

cc @ajosh0504 any thoughts on a generic name for this function?

Reply:
I'm partial to `doTheComplicatedMath()`, myself

Reply:
I would say something like `updatingOdds` is appropriate.

Reply:
My motivation behind the `bayesian` nomenclature was discoverability; previously we had just been referring to "odds" and it was quite difficult to discover the basis for the math with so few clues. As long as we have the reference to the bayesian update/inference in the JSDoc, I think we can freely change the name here without sacrificing information 👍